source/12_chapter_4.md

```diff
@@ -26,17 +26,22 @@ These are the results. In vitae odio at libero elementum fermentum vel iaculis e
 
 ## Discussion
 
-Figure \ref{ref_a_figure} shows how to add a figure. Donec ut lacinia nibh. Nam tincidunt augue et tristique cursus. Vestibulum sagittis odio nisl, a malesuada turpis blandit quis. Cras ultrices metus tempor laoreet sodales. Nam molestie ipsum ac imperdiet laoreet. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas.
+<!--
+Figures can be either referenced using @fig:mylabel, and can be auto-completed
+to **Fig. number: mylabel** by prefixing with @ for mid-sentence references and * for the start of sentences. See https://github.com/tomduck/pandoc-fignos
+-->
+
+Fig. @fig:my_fig shows how to add a figure. Donec ut lacinia nibh. Nam tincidunt augue et tristique cursus. Vestibulum sagittis odio nisl, a malesuada turpis blandit quis. Cras ultrices metus tempor laoreet sodales. Nam molestie ipsum ac imperdiet laoreet. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas.
 
 <!-- 
 Figures can be added with the following syntax:
-![my_caption \label{my_label}](source/figures/my_image.pdf){ width=50% }
+![main_text_caption](source/figures/my_image.pdf, "short_caption(optional)"){#fig:mylabel}{ width=50% }
```

Comment:
Here, there shouldn't be a comma between the image source and the short caption.

Reply:
Thanks for the support! I didn't realize it added links to the HTML, that's a nice bonus. 

I'll fix that annotation, and see about fixing the `template.tex`. To be honest, I had never used a template for the body and forgot to add it back in. I'll run it with the proposed changes to make sure it works as intended.

W.r.t. the `no-prefix-figure` error message, that occurs if any figure doesn't have the `{#number:label}` notation at the end. If I remember correctly, the pdf still compiled as expected, despite giving that error? I looked up this error on the fignos page, and it appears that adding the `--standalone` flag may fix this, which will add some peace of mind for end users. Will  try it out and get back to you.

Reply:
@dendrondal sorry for not getting around to dealing with this yet. if you are planning an update, i'll hold on for that and will then check and merge. thanks!

Reply:
@davidverweij Can you elaborate a bit more on your issue with caption prefixes? I've re-run the build both with and without the addition to the `template.tex` after adding the `template` flag to the makefile, and the output appears the same to me.